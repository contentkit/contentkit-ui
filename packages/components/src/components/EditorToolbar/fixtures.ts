export const anchorOrigin = {
  vertical: 'bottom',
  horizontal: 'center',
}

export const transformOrigin = {
  vertical: 'top',
  horizontal: 'center'
}