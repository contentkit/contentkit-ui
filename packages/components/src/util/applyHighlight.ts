import { Modifier, EditorState } from 'draft-js'

function applyHighlight (editorState, selection) {
  const contentState = Modifier.applyInlineStyle(
    editorState.getCurrentContent(),
    selection,
    'HIGHLIGHT'
  )
  return EditorState.push(
    editorState,
    contentState,
    'change-inline-style'
  )
}

export default applyHighlight
