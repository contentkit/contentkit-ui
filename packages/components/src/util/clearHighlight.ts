
import { Modifier, EditorState, SelectionState } from 'draft-js'

function clearHighlight (editorState, selection) {
  const contentState = Modifier.removeInlineStyle(editorState.getCurrentContent(), selection, 'HIGHLIGHT')
  const nextEditorState = EditorState.push(editorState, contentState, 'change-inline-style')
  return nextEditorState
  // return EditorState.forceSelection(nextEditorState, selection)
}

export default clearHighlight