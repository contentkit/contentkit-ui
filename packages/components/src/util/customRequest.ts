import { CustomRequestParams } from '../types'

async function customRequest ({ headers, file, action, onSuccess }: CustomRequestParams): Promise<void> {
  const formData = new window.FormData()
  for (let field in action.fields) {
    formData.append(field, action.fields[field])
  }
  formData.append('file', file)
  const resp = await fetch(action.url, {
    method: 'POST',
    body: formData,
    headers: headers
  })
  onSuccess(resp)
}

export default customRequest
