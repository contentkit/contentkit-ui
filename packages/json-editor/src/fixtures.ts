export enum Character {
  OPEN_SQUARE_BRACKET = '[',
  CLOSE_SQUARE_BRACKET = ']',
  OPEN_CURLY_BRACKET = '{',
  CLOSE_CURLY_BRACKET = '}',
  DOUBLE_QUOTE = '"'
}

export const CHARACTER_PAIRS = {
  [Character.OPEN_SQUARE_BRACKET]: Character.CLOSE_SQUARE_BRACKET,
  [Character.OPEN_CURLY_BRACKET]: Character.CLOSE_CURLY_BRACKET,
  [Character.DOUBLE_QUOTE]: Character.DOUBLE_QUOTE
}

export const OPEN_SCOPE_DELIMETERS = Object.keys(CHARACTER_PAIRS)
