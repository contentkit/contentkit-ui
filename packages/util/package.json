{
  "name": "@contentkit/util",
  "version": "0.0.62",
  "private": false,
  "description": "Util",
  "main": "lib/index.js",
  "keywords": [],
  "scripts": {
    "clean": "rimraf lib",
    "lint": "standard \"src/**/*.js\" --fix | snazzy",
    "build": "yarn run clean && tsc -p ./config",
    "watch": "tsc-watch -p ./config",
    "test": "jest"
  },
  "license": "MIT",
  "dependencies": {
    "left-indent": "^0.0.6"
  },
  "peerDependencies": {
    "draft-js": "0.11.6",
    "immutable": "3.7.6"
  },
  "standard": {
    "parser": "babel-eslint",
    "ignore": [
      "node_modules",
      "dist",
      ".cache"
    ]
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org/",
    "access": "public"
  },
  "devDependencies": {
    "babel-jest": "^24.9.0",
    "draft-js": "^0.11.6",
    "immutable": "3.7.6",
    "jest": "^24.9.0",
    "tsc-watch": "3.0.2",
    "typescript": "^3.7.4"
  },
  "jest": {
    "testMatch": [
      "**/src/**/*.test.js"
    ]
  },
  "gitHead": "af83b1d1596f29fc61c474034b39297155903e59"
}
