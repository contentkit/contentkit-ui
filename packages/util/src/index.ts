export * from './transactions'
export * from './fixtures'
export * from './utils'
