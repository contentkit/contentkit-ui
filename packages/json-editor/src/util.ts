
import { OPEN_SCOPE_DELIMETERS, CHARACTER_PAIRS } from './fixtures'

export const pad = (offset) => new Array(offset).fill(' ').join('')

export const DEFAULT_INDENT = pad(2)


export function isCharacterPair (left: string, right: string) {
  return OPEN_SCOPE_DELIMETERS.includes(left) && right === CHARACTER_PAIRS[left]
}

