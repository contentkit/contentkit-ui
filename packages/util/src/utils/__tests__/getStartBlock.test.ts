import { getStartBlock } from '@contentkit/util'
import { EditorState, SelectionState, ContentState, ContentBlock, CharacterMetadata } from 'draft-js'
import { List, Repeat } from 'immutable'
import { Block } from '../../fixtures'

const getEditorState = (blockArray, focusKey) => {
  const editorState = EditorState.createWithContent(
    ContentState.createFromBlockArray(blockArray)
  )
  return EditorState.acceptSelection(editorState,
    new SelectionState({
      anchorKey: focusKey,
      anchorOffset: 0,
      focusKey: focusKey,
      focusOffset: 0,
      isBackward: false
    })
  )
}

it('getStartBlock', () => {
  const blocksArray = [
    new ContentBlock({ key: 'A' }),
    new ContentBlock({
      key: 'B',
      type: Block.CODE,
      text: 'foo',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({
      key: 'C',
      type: Block.CODE,
      text: 'bar',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
    new ContentBlock({ key: 'D' })
  ]

  const editorState = getEditorState(blocksArray, 'C')
  const startBlock = getStartBlock(editorState)

  expect(startBlock).toMatchSnapshot()
  expect(startBlock.getKey()).toMatch('B')
})


it('getStartBlock', () => {
  const blocksArray = [
    new ContentBlock({
      key: 'A',
      type: Block.CODE,
      text: 'bar',
      characteList: List(Repeat(new CharacterMetadata(), 3))
    }),
  ]

  const editorState = getEditorState(blocksArray, 'A')
  const startBlock = getStartBlock(editorState)

  expect(startBlock).toMatchSnapshot()
  expect(startBlock.getKey()).toMatch('A')
})
