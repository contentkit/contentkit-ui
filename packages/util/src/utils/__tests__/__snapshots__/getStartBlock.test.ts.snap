// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStartBlock 1`] = `
Immutable.Record {
  "key": "B",
  "type": "code-block",
  "text": "foo",
  "characterList": Immutable.List [
    Immutable.Record {
      "style": Immutable.OrderedSet [],
      "entity": null,
    },
    Immutable.Record {
      "style": Immutable.OrderedSet [],
      "entity": null,
    },
    Immutable.Record {
      "style": Immutable.OrderedSet [],
      "entity": null,
    },
  ],
  "depth": 0,
  "data": Immutable.Map {},
}
`;

exports[`getStartBlock 2`] = `
Immutable.Record {
  "key": "A",
  "type": "code-block",
  "text": "bar",
  "characterList": Immutable.List [
    Immutable.Record {
      "style": Immutable.OrderedSet [],
      "entity": null,
    },
    Immutable.Record {
      "style": Immutable.OrderedSet [],
      "entity": null,
    },
    Immutable.Record {
      "style": Immutable.OrderedSet [],
      "entity": null,
    },
  ],
  "depth": 0,
  "data": Immutable.Map {},
}
`;
